### ISSUE TITLE
Error open (<:unix) on '.../JSON/Schema/Modern/Document/OpenAPI/oas/schema.json': No such file or directory

### ISSUE THREAD

**Issue by eserte** [karenetheridge/JSON-Schema-Modern-Document-OpenAPI#15]
The test suite fails on some of my smoker systems:
```
    #   Failed test 'no exception when the document itself is provided'
    #   at t/openapi-constructor.t line 56.
    #          got: 'Error open (<:unix) on '/home/cpansand/.cpan/build/2021112623/JSON-Schema-Modern-Document-OpenAPI-0.003-0/blib/lib/auto/JSON/Schema/Modern/Document/OpenAPI/oas/schema.json': No such file or directory at /home/cpansand/.cpan/build/2021112623/JSON-Schema-Modern-Document-OpenAPI-0.003-0/blib/lib/JSON/Schema/Modern/Document/OpenAPI.pm line 151.
    # '
    #     expected: undef
...
t/openapi-constructor.t .. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
    # No tests run!

#   Failed test 'No tests run for subtest "error handling"'
#   at t/validate_request.t line 620.
Error open (<:unix) on '/home/cpansand/.cpan/build/2021112623/JSON-Schema-Modern-Document-OpenAPI-0.003-0/blib/lib/auto/JSON/Schema/Modern/Document/OpenAPI/oas/dialect/base.schema.json': No such file or directory at /home/cpansand/.cpan/build/2021112623/JSON-Schema-Modern-Document-OpenAPI-0.003-0/blib/lib/JSON/Schema/Modern/Document/OpenAPI.pm line 151.
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 1.
t/validate_request.t ..... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1/1 subtests 
```

Pass:fail ratio on my smokers is currently 30:4. The failing configurations look like to be system perls on Debian (or Debian-like) systems.

**Comment by karenetheridge** [comment]
This seems to be due to a strange interaction between File::ShareDir and cpanm -- I can't reproduce it locally at all but it seems that sometimes, when downloading/installing a module with cpanm, some of the sharedir files are missing in blib.

**Comment by eserte** [comment]
I can reproduce it using docker and the following Dockerfile
```
# docker build -t perl-test . && docker run perl-test
FROM ubuntu:20.04
RUN apt update
RUN apt install -y perl-modules make gcc
RUN cpan -J
CMD cpan -t ETHER/JSON-Schema-Modern-Document-OpenAPI-0.003.tar.gz
```

**Comment by karenetheridge** [comment]
I mixed this issue up with a strange issue I observed in another one of my dists, where some of the sharedir files were not being properly added to blib by Build.PL.

The issue here was simply that I left off `use Test::File::ShareDir` in a few tests, so running tests on any machine where the distribution hadn't already been installed would fail -- and fixed in v0.004.
